agent/engine/docker_task_engine.go

```diff
@@ -657,6 +663,35 @@ func (engine *DockerTaskEngine) pullContainer(task *api.Task, container *api.Con
 	return engine.serialPull(task, container)
 }
 
+// useLocalImageCache returns true if local image cache should be used, or return false to continue
+// pulling image, by inspecting the agent pull behavior variable defined in config. The caller has
+// to make sure the container passed in is not an internal container.
+func (engine *DockerTaskEngine) useLocalImageCache(agentPullBehavior config.AgentPullBehaviorType,
+	container *api.Container,
+	taskArn string) bool {
+	// If the agent pull behavior is never, we don't try to pull the image,
+	// try to use local image cache instead.
+	if agentPullBehavior == config.NeverAgentPullBehavior {
```

Comment:
minor: this'd be more readable with a `switch` and cases.

Reply:
OK